source/logrepl/copyto_test.go

```diff
@@ -0,0 +1,175 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logrepl
+
+import (
+	"context"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/conduitio/conduit-connector-postgres/test"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/matryer/is"
+)
+
+func TestCopyTo(t *testing.T) {
+	is := is.New(t)
+	ctx := context.Background()
+	conn := test.ConnectSimple(ctx, t, test.RepmgrConnString)
+
+	_, err := conn.Exec(ctx, `create temporary table foo( a int2, b int4, c int8, d varchar, e text, f date, g json)`)
+	is.NoErr(err)
+
+	_, err = conn.Exec(context.Background(), `insert into foo values (0, 1, 2, 'abc	', 'efg', '2000-01-01', '{"abc":"def","foo":"bar"}')`)
+	is.NoErr(err)
+
+	_, err = conn.Exec(context.Background(), `insert into foo values (3, null, null, null, null, null, '{"foo":"bar"}')`)
+	is.NoErr(err)
+
+	w, err := NewCopyDataWriter(ctx, conn, Config{TableName: "foo"})
+	is.NoErr(err)
+
+	go w.Copy(ctx, conn)
+
+	now := time.Now()
+	count := 0
+	for count < 2 {
+		rec, err := w.Next(ctx)
+		is.NoErr(err)
+		is.True(rec.CreatedAt.After(now))
+		count++
+	}
+
+	<-w.done
+	is.NoErr(w.Teardown(ctx))
+}
+
+func TestCopyWriter_Copy(t *testing.T) {
+	is := is.New(t)
+	ctx := context.Background()
+	conn := test.ConnectSimple(ctx, t, test.RepmgrConnString)
+
+	_, err := conn.Exec(ctx, `create temporary table foo( a int2, b int4, c int8, d varchar, e text, f date, g json)`)
+	is.NoErr(err)
+
+	_, err = conn.Exec(context.Background(), `insert into foo values (0, 1, 2, 'abc	', 'efg', '2000-01-01', '{"abc":"def","foo":"bar"}')`)
+	is.NoErr(err)
+
+	_, err = conn.Exec(context.Background(), `insert into foo values (3, null, null, null, null, null, '{"foo":"bar"}')`)
+	is.NoErr(err)
+
+	tx, err := conn.BeginTx(ctx, pgx.TxOptions{
+		IsoLevel:   "REPEATABLE",
+		AccessMode: "READ",
+	})
+	is.NoErr(err)
+	defer is.NoErr(tx.Commit(ctx))
+
+	w, err := NewCopyDataWriter(ctx, tx.Conn(), Config{TableName: "foo"})
+	is.NoErr(err)
+
+	go w.Copy(ctx, conn)
+
+	now := time.Now()
+	count := 0
+	for count < 2 {
+		rec, err := w.Next(ctx)
+		is.NoErr(err)
+		is.True(rec.CreatedAt.After(now))
+		count++
+	}
+
+	<-w.done
+	is.NoErr(w.Teardown(ctx))
+}
+
+func TestCopyDataWriter_Next(t *testing.T) {
+	is := is.New(t)
+	type args struct {
+		config Config
+	}
+	tests := []struct {
```

Comment:
Friendly tip: don't start with a data-driven test until you know you need one. We have a single test case, this could be simplified.